rules/windows/file/file_access/file_access_win_powershell_console_history_file_access.yml

```diff
@@ -0,0 +1,29 @@
+title: PowerShell Console History File Access
+id: 4d052abb-0bee-44f3-9aa0-8dab6b1bd349
+related:
+    - id: f4ff7323-b5fc-4323-8b52-6b9408e15788
+      type: derived
+status: experimental
+description: Detects access to the PowerShell console history file (ConsoleHost_history.txt), possibly to retrieve plaintext passwords used in PowerShell commands.
+references:
+    - https://0xdf.gitlab.io/2018/11/08/powershell-history-file.html
+author: Luc Génaux
+date: 2025-04-03
+tags:
+    - attack.credential-access
+    - attack.t1552.001
+logsource:
+    product: windows
+    category: file_access
+    definition: 'Requirements: Microsoft-Windows-Kernel-File ETW provider'
+detection:
+    selection:
+        FileName|endswith: 'PSReadLine\ConsoleHost_history.txt'
```

Comment:
Powershell, AV, Windows itself and many other processes will request access to this file.

The filters have to be more strict or the rule will generate a lot of FPs